packages/router/src/adapters/contract/contract.ts

```diff
@@ -18,6 +28,76 @@ export const getContractAddress = (chainId: number): string => {
 export const getTxManagerInterface = () =>
   new Interface(TransactionManagerArtifact.abi) as TTransactionManager["interface"];
 
+export const prepareSanitationCheck = async (
+  chainId: number,
+  nxtpContractAddress: string,
+  invariantTransactionData: InvariantTransactionData,
+) => {
```

Comment:
Do we really want to add in an onchain transaction for each active transaction loop? This is going to make our provider calls skyrocket and seems an inelegant solution to problems with the subgraph

Reply:
It's one provider call per tx per loop which shouldnt be that much but I agree. We may not even need this anymore.